src/cljs/rems/fields.cljs

```diff
@@ -142,24 +142,30 @@
   (.. event -target -value))
 
 (defn text-field
-  [{:keys [validation on-change] :as opts}]
+  [{:keys [validation on-change collapse-info info-text] :as opts}]
   (let [placeholder (localized (:field/placeholder opts))
         value (:field/value opts)
         optional (:field/optional opts)
         max-length (:field/max-length opts)]
     [field-wrapper opts
-     [:input.form-control {:type "text"
-                           :id (field-name opts)
-                           :name (field-name opts)
-                           :placeholder placeholder
-                           :required (not optional)
-                           :aria-invalid (when validation true)
-                           :aria-describedby (when validation
-                                               (str (field-name opts) "-error"))
-                           :max-length max-length
-                           :class (when validation "is-invalid")
-                           :value value
-                           :on-change (comp on-change event-value)}]]))
+     [:<>
+      (if collapse-info [collapse
+                         (field-name opts)
+                         (str "some text")
+                         focus-when-collapse-opened
+                         [:span (str info-text)]])
```

Comment:
There is no need of `str` here if you implement then `if/when` without having a separate `collapse-info`.

Reply:
Also no need for the `:span` either. One could give a more complicated info-text and these are preventing it.

Reply:
ok, I see your point